apps/client/src/app/views/HomeView.tsx

```diff
@@ -16,101 +30,152 @@ export const HomeView: FC = () => {
+        !priceConsumerContract
+      ) {
+        return;
+      }
 
-  useEffect(() => {
-    async function updateMankPrice() {
-      if (!privateSaleContract || !priceConsumerContract) return;
-      const mankPriceInWei = await privateSaleContract.getTokenCostInWei();
+      const getMankBalance = async () => {
+        const balance = await mankTokenContract.balanceOf(currentAddress);
 
-      const ethPriceInUsd = await priceConsumerContract.getEthUsdPrice();
+        setBalance(balance);
+      };
 
-      const mankPriceInWeiInUsd = mankPriceInWei.mul(ethPriceInUsd);
+      const getRedeemableTokens = async () => {
+        const redeemableTokens =
+          await privateSaleContract.getMyRedeemableTokens();
 
-      setMankPrice(mankPriceInWeiInUsd);
-      setMankPriceInWei(mankPriceInWei);
-    }
-    updateMankPrice();
-  }, [priceConsumerContract, privateSaleContract]);
+        setRedeemableTokens(redeemableTokens);
+      };
 
-  useEffect(() => {
-    async function updatePurchaseUnits() {
-      if (!privateSaleContract) return;
-      const unit = await privateSaleContract.purchaseUnit();
+      const getMankPrices = async () => {
+        const mankPriceInWei = await privateSaleContract.getTokenCostInWei();
 
-      const maxPurchase = +(await privateSaleContract.maxPurchase()).toString();
+        setMankPriceInWei(mankPriceInWei);
 
-      console.log(maxPurchase);
+        const ethPriceInUsd = await priceConsumerContract.getEthUsdPrice();
 
-      setPurchaseUnit(+formatEther(unit));
-      setMaxPurchase(maxPurchase);
-    }
-    updatePurchaseUnits();
-  }, [privateSaleContract]);
+        const mankPriceInWeiInUsd = mankPriceInWei.mul(ethPriceInUsd);
 
-  useEffect(() => {
-    async function updateMinPurchase() {
-      if (!privateSaleContract) return;
+        setMankPrice(mankPriceInWeiInUsd);
+      };
+
+      const getPurchaseUnit = async () => {
+        const unit = await privateSaleContract.purchaseUnit();
+
+        setPurchaseUnit(unit);
+      };
+
+      const getMinPurchase = async () => {
+        const minPurchase = await privateSaleContract.minPurchase();
 
-      const minPurchase = await privateSaleContract.minPurchase();
-      setMinPurchase(minPurchase);
+        setMinPurchase(minPurchase);
+
+        setMankToBuy(minPurchase.toNumber());
+      };
+
+      const getMaxPurchase = async () => {
+        const maxPurchase = await privateSaleContract.maxPurchase();
+
+        setMaxPurchase(maxPurchase);
+      };
+
+      try {
+        await Promise.all([
+          getMankBalance(),
+          getRedeemableTokens(),
+          getMankPrices(),
+          getPurchaseUnit(),
+          getMinPurchase(),
+          getMaxPurchase(),
+        ]);
+      } catch (err) {
+        setFailedFetchingData(true);
+        return;
+      }
+      setFailedFetchingData(false);
     }
-    updateMinPurchase().catch(console.error);
-  }, [privateSaleContract]);
+    getInitialData();
```

Comment:
```suggestion
    getInitialData().catch(console.error);
```
So that we don't lose the error